### PR TITLE
feat: add opentelemetry example for alloy

### PR DESCRIPTION
**Pull request by sunng87**
This patch reuse alloy example to demo opentelemetry data ingestion.

For more information, see the README.

### CODE REVIEW

File: grafana-alloy/README.md [hunk 1]
```diff
@@ -67,19 +73,39 @@ The topology is illustrated in this diagram.
 ```mermaid
 flowchart LR
   greptimedb[(GreptimeDB)]
-  alloy[Alloy]
   grafana[Grafana]
 
-  alloy --> greptimedb
+  subgraph alloy[Alloy]
+    prometheus_exporter_unix --> prometheus_relabel
+    prometheus_relabel --> prmetheus_remote_write
```
Comment (daviderli614) [inline grafana-alloy/README.md:80]:
```suggestion
    prometheus_relabel --> prometheus_remote_write
```

File: grafana-alloy/README.md [hunk 2]
```diff
@@ -67,19 +73,39 @@ The topology is illustrated in this diagram.
...
+
+    prometheus_relabel --> otelcol_receiver_prometheus
+    otelcol_receiver_prometheus --> otelcol_processor_transform
+    otelcol_processor_transform --> otelcol_exporter_otlphttp
+  end
+
+  prmetheus_remote_write --> |PRW| greptimedb
```
Comment (daviderli614) [inline grafana-alloy/README.md:87]:
```suggestion
  prometheus_remote_write --> |PRW| greptimedb
```

**Comment by daviderli614** [comment]
I tried to run the demo and got an error：
```
git clone https://github.com/GreptimeTeam/demo-scene.git
cd demo-scene/grafana-alloy
```
```
docker compose up
services.envsubst.env_file.0 must be a string
```
image

**Comment by sunng87** [comment]
@dedlockdave according to https://docs.docker.com/compose/how-tos/environment-variables/set-environment-variables/#additional-information-1, it requires a minimal version of docker compose. I will add to readme.

File: grafana-alloy/README.md [hunk 3]
```diff
@@ -67,19 +73,39 @@ The topology is illustrated in this diagram.
...
+In this example, we generate metrics data from Alloy's built-in data source
+called `prometheus.exporter.unix` and export the data to both a Prometheus
+remote write sink and an OpenTelemetry-compatible collector, which is
+GreptimeDB.
+
 ## Run in GreptimeCloud
 
 By default, this example writes data into a GreptimeDB instance within the
 docker compose. It's also possible to write to your own GreptimeCloud instance
 by creating a `greptime.env` file from our sample `greptime.env.sample` and
```
Comment (daviderli614) [inline grafana-alloy/README.md:107]: Added a link to greptime.env.sample, it looks like [`greptime.env.sample`](./greptime.env.sample) ?

Comment (sunng87) [reply grafana-alloy/README.md:107]: I think it's ok. The idea is to let user copy the sample and rename

**Comment by sunng87** [comment]
@dedlockdave I was going to invite David but Li Yang's id was quite similar @daviderli614. Thank you for reviewing this and it has been quite helpful.

**Review by dedlockdave — APPROVED** [review]

**Pull request merged**